crates/next-api/src/next_server_nft.rs

```diff
@@ -0,0 +1,340 @@
+        .as_ref()
+        .and_then(Value::as_object)
+    {
+        for (glob_pattern, exclude_patterns) in output_file_tracing_excludes {
+            // Check if the route matches the glob pattern
+            let glob = Glob::new(RcStr::from(glob_pattern.clone()), Default::default()).await?;
+            if glob.matches("next-server")
+                && let Some(patterns) = exclude_patterns.as_array()
+            {
+                for pattern in patterns {
+                    if let Some(pattern_str) = pattern.as_str() {
+                        additional_ignores.insert(pattern_str);
+                    }
+                }
+            }
+        }
+    }
+
+    let server_ignores_glob = [
+        "**/node_modules/react{,-dom,-server-dom-turbopack}/**/*.development.js",
+        "**/*.d.ts",
+        "**/*.map",
+        "**/next/dist/pages/**/*",
+        "**/next/dist/compiled/next-server/**/*.dev.js",
+        "**/next/dist/compiled/webpack/*",
+        "**/node_modules/webpack5/**/*",
+        "**/next/dist/server/lib/route-resolver*",
+        "**/next/dist/compiled/semver/semver/**/*.js",
+        "**/next/dist/compiled/jest-worker/**/*",
+        // Turbopack doesn't support AMP
+        "**/next/dist/compiled/@ampproject/toolbox-optimizer/**/*",
+        // -- The following were added for Turbopack specifically --
+        // client/components/use-action-queue.ts has a process.env.NODE_ENV guard, but we can't set that due to React: https://github.com/vercel/next.js/pull/75254
+        "**/next/dist/next-devtools/userspace/use-app-dev-rendering-indicator.js",
+        // client/components/app-router.js has a process.env.NODE_ENV guard, but we can't set that.
+        "**/next/dist/client/dev/hot-reloader/app/hot-reloader-app.js",
+        // server/lib/router-server.js doesn't guard this require:
+        "**/next/dist/server/lib/router-utils/setup-dev-bundler.js",
+        // server/next.js doesn't guard this require
+        "**/next/dist/server/dev/next-dev-server.js",
+        // next/dist/compiled/babel* pulls in this, even we never actually transpile with  Babel
+        "**/next/dist/compiled/browserslist/**",
+    ]
+    .into_iter()
+    .chain(additional_ignores)
+    .chain(
+        if has_next_support {
+            Some(["**/node_modules/sharp/**/*", "**/@img/sharp-libvips*/**/*"]).into_iter()
+        } else {
+            None.into_iter()
+        }
+        .flatten(),
+    )
+    .chain(if has_next_support {
+        // only ignore image-optimizer code when
+        // this is being handled outside of next-server
+        Some("**/next/dist/server/image-optimizer.js").into_iter()
+    } else {
+        None.into_iter()
+    })
+    .chain(
+        if is_standalone {
+            None.into_iter()
+        } else {
+            Some(["**/*/next/dist/server/next.js", "**/*/next/dist/bin/next"]).into_iter()
+        }
+        .flatten(),
```

Comment:
Nit: I'd rather just use `iter::empty()` + `<[&str; 2] as IntoIterator>::into_iter` + `Either` than abusing `Option` and adding an extra `.flatten` into the end here.